tests/api/v1/generatorTemplates/post.js

```diff
@@ -0,0 +1,119 @@
+const constants = require('../../../../api/v1/constants');
+const tu = require('../../../testUtils');
+const u = require('./utils');
+const path = '/v1/generatorTemplates';
+const GeneratorTemplate = tu.db.GeneratorTemplate;
+const expect = require('chai').expect;
+const ZERO = 0;
+
+describe('tests/api/v1/generatorTemplates/post.js > ', () => {
+  let token;
+  const generatorTemplate = u.getGeneratorTemplate();
+  before((done) => {
+    tu.createToken()
+    .then((returnedToken) => {
+      token = returnedToken;
+      done();
+    })
+    .catch(done);
+  });
+  afterEach(u.forceDelete);
+  after(tu.forceDeleteUser);
+
+  it('simple post OK', (done) => {
+    api.post(path)
+    .set('Authorization', token)
+    .send(generatorTemplate)
+    .expect(constants.httpStatus.CREATED)
+    .end((err, res) => {
+      if (err) {
+        return done(err);
+      }
+      expect(res.body.apiLinks).to.be.an('Array');
+      expect(res.body.name).to.include(generatorTemplate.name);
+      expect(res.body.id).to.not.equal(undefined);
+      expect(res.body).to.have.any.keys(Object.keys(generatorTemplate));
+      return done();
+    });
+  });
+
+  it('simple post without requred fields', (done) => {
+    const _generatorTemplate = JSON.parse(JSON.stringify(generatorTemplate));
+
+    delete _generatorTemplate.name;
+    delete _generatorTemplate.author;
+    delete _generatorTemplate.connection;
+    delete _generatorTemplate.transform;
+    delete _generatorTemplate.isPublished;
+
+    api.post(path)
+    .set('Authorization', token)
+    .send(_generatorTemplate)
+    .expect(constants.httpStatus.CREATED)
+    .end((err, res) => {
+      if (!err) {
+        return done('Expecting "Schema Validation Failed" error');
+      }
+      const errorArray = JSON.parse(res.text).errors;
+      expect(errorArray.length).to.equal(5);
+      expect(errorArray[ZERO].type).to.equal('SCHEMA_VALIDATION_FAILED');
+      return done();
+    });
+  });
+
+  describe('post duplicate fails > ', () => {
+    beforeEach((done) => {
+      GeneratorTemplate.create(generatorTemplate)
+      .then(() => done())
+      .catch(done);
+    });
+
+    it('with identical name', (done) => {
+      api.post(path)
+      .set('Authorization', token)
+      .send(generatorTemplate)
+      .expect(constants.httpStatus.FORBIDDEN)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res.body.errors[ZERO].type)
+          .to.equal(tu.uniErrorName);
+        done();
+      });
+    });
+
+    it('with case different name', (done) => {
+      generatorTemplate.name = generatorTemplate.name.toLowerCase();
+      api.post(path)
+      .set('Authorization', token)
+      .send(generatorTemplate)
+      .expect(constants.httpStatus.FORBIDDEN)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res.body.errors[ZERO].type)
+          .to.equal(tu.uniErrorName);
```

Comment:
when an expect(...) like this all fits on one line in under 80 chars, don't break it into two lines